crates/forge/src/test_filter.rs

```diff
@@ -76,16 +91,26 @@ impl TestsFilter {
             }
         };
 
+        if self.last_failed_filter {
+            cases = match get_cached_failed_tests_names(&self.cache_dir_path)? {
+                Some(result) => cases
+                    .into_iter()
+                    .filter(|tc| result.iter().any(|name| name == &tc.name))
+                    .collect(),
+                None => cases,
+            }
+        }
```

Comment:
This will allow `--last-failed` filtering to be combined with other filters. I'm not sure if this is a good behavior. Perhaps we should return early here? Or have it as an entirely separate filter.

Reply:
cc @piotmag769

Reply:
but what is wrong with combining with other filters?